### ISSUE TITLE
Tests assume a specific logging configuration

### ISSUE THREAD

**Issue by julen** [translate/pootle#4448]
Tests seem to assume a specific logging configuration. Anyone having their own settings, they will be used over defaults, and tests won't pass.

Tests should specify their own logging settings if they require so.

**Comment by dwaynebailey** [comment]
@julen I don't think this is actually the problem.  We don't seem to be able to catch the logs from the action logger to be able to test them.

So would be helpful if you could explain how your logs are different from default for the action logger.

The test when run against Django 1.9 also don't pick up the action logger while in 1.8 master they do.  Yet on your 1.8 install they don't.  So hopefully how your settings are vs master could tell us something about the general problem.

**Comment by julen** [comment]
I don't have the `console` handler for the `action` logger and adding it makes a difference.

So the problem is tests are reading the output of the logger from `stderr`, thus forcing a specific handler to be configured. Ideally this information should be read from logging directly instead of from a specific handler (there are tools to assist on this already, such as [pytest-capturelog](https://pypi.python.org/pypi/pytest-capturelog/) or [testfixtures](http://testfixtures.readthedocs.org/en/latest/logging.html)).

**Comment by dwaynebailey** [comment]
@julen thanks for confirming the difference.  And nope pytest-capturelog isn't solving this as I've not been able to get it to capture the action logger no matter how I ask it.

**Comment by julen** [comment]
Well, I didn't try, but any of the linked approaches feel saner than the current one TBH, especially because they are independent of any configurations.

**Comment by julen** [comment]
Also related to tests assuming certain settings (but not enforcing them), there's also this failure:

```pytb
__________________________________________________________________________ test_update_tmserver_nosetting ___________________________________________________________________________
tests/commands/update_tmserver.py:22: in test_update_tmserver_nosetting
    call_command('update_tmserver')
../../../.virtualenvs/pootle/lib/python2.7/site-packages/django/core/management/__init__.py:120: in call_command
    return command.execute(*args, **defaults)
../../../.virtualenvs/pootle/lib/python2.7/site-packages/django/core/management/base.py:445: in execute
    output = self.handle(*args, **options)
pootle/apps/pootle_app/management/commands/update_tmserver.py:357: in handle
    self._parse_translations(**options))
../../../.virtualenvs/pootle/lib/python2.7/site-packages/elasticsearch/helpers/__init__.py:182: in bulk
    for ok, item in streaming_bulk(client, actions, **kwargs):
../../../.virtualenvs/pootle/lib/python2.7/site-packages/elasticsearch/helpers/__init__.py:110: in streaming_bulk
    for action, data in chunk:
pootle/apps/pootle_app/management/commands/update_tmserver.py:250: in _parse_translations
    sys.exit()
E   SystemExit
------------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------------
Last indexed revision = 1003776.0
No translations to index
```

**Comment by julen** [comment]
I have renamed the title of the issue so it describes better the broader scope of the problem.
